delphi-frontend/src/main/java/au/com/integradev/delphi/reporting/DelphiIssueBuilderImpl.java

```diff
@@ -187,18 +229,69 @@ public void report() {
     }
 
     if (flows != null) {
-      for (List<DelphiCheckContext.Location> flow : flows) {
+      for (List<Location> flow : flows) {
         newIssue.addFlow(
             flow.stream()
                 .map(location -> createNewIssueLocation(inputFile, newIssue, location))
                 .collect(Collectors.toList()));
       }
     }
 
+    if (quickFixes != null && context.runtime().getProduct() == SonarProduct.SONARLINT) {
+      for (QuickFix quickFix : quickFixes) {
+        newIssue.addQuickFix(createNewQuickFix(delphiFile, newIssue, quickFix));
+      }
+    }
+
     newIssue.save();
     reported = true;
   }
 
+  private static NewQuickFix createNewQuickFix(
+      DelphiInputFile delphiFile, NewIssue newIssue, QuickFix quickFix) {
+    NewQuickFix newQuickFix = newIssue.newQuickFix().message(quickFix.getDescription());
+
+    NewInputFileEdit edit = newQuickFix.newInputFileEdit().on(delphiFile.getInputFile());
+
+    Supplier<DelphiFileStream> fileStreamSupplier =
+        Suppliers.memoize(
+            () -> {
+              try {
+                return new DelphiFileStream(
+                    delphiFile.getSourceCodeFile().getAbsolutePath(),
+                    delphiFile.getSourceCodeFileEncoding());
+              } catch (IOException e) {
+                throw new UncheckedIOException(e);
+              }
+            });
+
+    quickFix.getEdits().stream()
+        .map(QuickFixEditImpl.class::cast)
+        .map(e -> e.toTextEdits(fileStreamSupplier))
+        .flatMap(Collection::stream)
+        .map(
+            textEdit ->
```

Comment:
Hate it when lambdas get so nested and gnarly-looking. Maybe a separate mapping function?